### PR TITLE
Werewolf dash

### PR DESCRIPTION
**Pull request by MrJackey**
## Summary
Fixes # 14
Player can dash with 'shift' or 'left/right trigger'.
Animation when dashing.
Some minor refactoring of other code.

DO TEST IT THOROUGHLY!!

## Visualisation
NyvTiG4MLi

### CODE REVIEW

**Review by Addehman — APPROVED** [review]
Feels great!

File: Assets/Scripts/Player/PlayerController.cs [hunk 1]
```diff
@@ -143,4 +183,53 @@ public class PlayerController : MonoBehaviour {
...
+		doAirJump = false;
+		doGravity = true;
+	}
+
+	private bool CheckDashInput() {
+		if (Mathf.Abs(Input.GetAxisRaw("Dash")) > 0) {
+			bool oldState = dashHeld;
+
+			dashHeld = true;
+			return oldState == false;;
```
Comment (smeas) [inline Assets/Scripts/Player/PlayerController.cs]:
```suggestion
			return oldState == false;;;
```

File: Assets/Scripts/Player/PlayerController.cs [hunk 2]
```diff
@@ -111,13 +136,28 @@ public class PlayerController : MonoBehaviour {
 		else if (doAirJump)
 			Jump(airJumpCurve, airJumpEndTime);
 
-		if (!doJump || !doAirJump)
-			rb2D.velocity = new Vector2(velocity.x, rb2D.velocity.y + gravity * Time.deltaTime);
+		if (doDash)
+			Dash();
+		else if (!allowDashReset) {
```
Comment (smeas) [inline Assets/Scripts/Player/PlayerController.cs]: Braces?
```suggestion
		if (doDash) {
			Dash();
		} else if (!allowDashReset) {
```

**Review by smeas — CHANGES_REQUESTED** [review]
Gravity scale is back??
image

**Comment by smeas** [comment]
The dash distance feels a bit short.

**Review by smeas — APPROVED** [review]
+1, It's not pretty, but it works!

**Pull request merged**